docs/src/cookbooks/cookbook.md

```diff
@@ -157,6 +158,39 @@ See the [DataView cookbook](dataview#how-do-i-connect-a-subset-of-bundle-fields)
 
 ## Vectors and Registers
 
+### Can I make a 2D or 3D Vector?
+
+Yes. Using `VecInit` you can make Vectors that hold Vectors of Chisel types. Methods `fill` and `tabulate` make these multi-dimensional Vectors.
+
+```scala mdoc:silent:reset
+import chisel3._
+
+class MyBundle extends Bundle {
+  val foo = UInt(4.W)
+  val bar = UInt(4.W)
+}
+
+class Foo extends Module {
+  
+  val twoDVec = VecInit.fill(2,3)(5.U)
+  val threeDVec = VecInit.fill(1,2,4)(new MyBundle)
+
+  val indexTiedVec = VecInit.tabulate(2,2){ (x, y) => (x+y).asUInt }
```

Comment:
```suggestion
  val indexTiedVec = VecInit.tabulate(2, 2){ (x, y) => (x + y).U }
```
While `.asUInt` is defined on `Int`, we typically use just `.U`.

Also general Scala style is to have spaces around operators and after commas